app/controllers/assignments_controller.rb

```diff
@@ -550,6 +550,21 @@ def download_sample_starter_files
     send_file zip_path, filename: zip_name
   end
 
+  # Downloads a zip file containing all the information and settings about an assignment
+  def download_config_files
+    assignment = Assignment.find(params[:id])
+
+    zip_name = "#{assignment.short_identifier}-config-files.zip"
+    zip_path = File.join('tmp', zip_name)
+
+    FileUtils.rm_f(zip_path)
+
+    Zip::File.open(zip_path, create: true) do |zip_file|
+      assignment.configuration_file_entries.each { |entry| entry.add_files_to_zip_file(zip_file) }
```

Comment:
I'm glad you're looking at other bits fo the code to see how we do things like create zip files in other places. This is a great way to figure stuff out.

One thing to be careful of is that you don't just copy and paste without understanding what the code is doing. In this case, it looks like you were looking at the code for writing a starter file group to a zip file. You're using the `add_files_to_zip_file` method which is a method for the StarterFileEntry model, this method will not be available when you are writing config files (since config files are not StarterFileEntry models).

For now, please just change this to:

```ruby
Zip::File.open(zip_path, create: true)
```

which will create an empty zip file. Then we can discuss later how to go about adding files to a zip file we're creating.